newIDE/app/src/InstancesEditor/InstancesAdder.js

```diff
@@ -39,24 +42,52 @@ const roundPositionsToGrid = (
 export default class InstancesAdder {
   _instances: gdInitialInstancesContainer;
   _temporaryInstances: Array<gdInitialInstance>;
+  _onAdditionalServiceComplete: (infobarDetails: InfoBarDetails) => void;
   _options: Object;
   _zOrderFinder = new gd.HighestZOrderFinder();
 
-  constructor({ instances, options }: Props) {
+  constructor({ instances, options, onAdditionalServiceComplete }: Props) {
     this._instances = instances;
     this._options = options;
     this._temporaryInstances = [];
+    this._onAdditionalServiceComplete = onAdditionalServiceComplete;
   }
 
   setOptions(options: Object) {
     this._options = options;
   }
 
+  onAddInstanceAdditionalservice = (
```

Comment:
As explained in the other comment, let's entirely remove this from this class. Just pass to the class a function to be called when an instance is added. The parent will figure out what to do, but as `InstancesAdder` you:
- don't know what will happen
- and you don't want to know! ;) Your job is to put instances on the scenes, not to run the work of others.